File: vue.jsonforms/src/entities/providers-and-introducers/ProviderIntroducerDetails.ts
import { ContactNumber } from "../ContactNumber";
import {ProductType} from "@/entities/org-structure/ProductType";

export class ProviderIntroducerDetails {
  public name?: string;
  public foreName?: string;
  public lastName?: string;
  public companyNumber?: string;
  public fcaFirmRefNo?: string;
  public praFirmRefNo?: string;
  public registeredAddress?: string;
  public tradingAddress?: string;
  public homeAddress?: string;
  public emailAddress?: string;
  public contactNumber?: ContactNumber;
  public website?: string;
  public tradingName?: string;
  public notes?: string;
  public praAuthorised?: boolean;
  public isTradingSameAsRegisteredAddress: boolean = false;
  public contactNumberDisplay?: string = "";
  public isCompany: boolean = true;
  public productType?: ProductType[] = [];
}

File: CSharp.Functions/Common/Entities/FirmDetails/ProvidersAndIntroducers/ProviderIntroducerDetails.cs
using Common.Entities.FirmDetails;

namespace Common.Entities
{
    public class ProviderIntroducerDetails
    {
        public string? Name { get; set; }
        public string? ForeName { get; set; }
        public string? LastName {  get; set; }
        public string? CompanyNumber { get; set; }
        public string? FcaFirmRefNo { get; set; }
        public string? PraFirmRefNo { get; set; }
        public string? RegisteredAddress { get; set; }
        public string? TradingAddress { get; set; }
        public string? HomeAddress { get; set; }
        public string? EmailAddress { get; set; }
        public ContactNumber? ContactNumber { get; set; }
        public string? Website { get; set; }
        public string? TradingName { get; set; }
        public string? Notes { get; set; }
        public bool? PraAuthorised { get; set; }
        public bool IsTradingSameAsRegisteredAddress { get; set; }
        public string? ContactNumberDisplay { get; set; }
        public bool IsCompany { get; set; }
        public List<ProductType> ProductType { get; set; } = new();
    }
}
